doc/source/user_guide/io.rst

```diff
@@ -272,9 +272,12 @@ parse_dates : boolean or list of ints or names or list of lists or dict, default
 
   .. note::
      A fast-path exists for iso8601-formatted dates.
+
+  .. note::
+   The ``infer_datetime_format`` is deprecated and will be removed in the future versions. 
 infer_datetime_format : boolean, default ``False``
```

Comment:
check here for how to mark as deprecated

https://github.com/pandas-dev/pandas/blob/eb226bdfbe24f4be49c42b4a29d56cca554e921e/doc/source/user_guide/io.rst#L157-L162